### PR TITLE
[kitchen] Update dogstatsd install script to use keys.datadoghq.com

### PR DESCRIPTION
**Pull request by bkabrda**
### What does this PR do?

Adapt the install script used in kitchen tests to install dogstatds to use keys.datadoghq.com for both RPM and APT keys.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/kitchen/site-cookbooks/dd-agent-install-script/files/dogstatsd_install_script.sh [hunk 1]
```diff
@@ -196,19 +239,42 @@ if [ "$OS" = "RedHat" ]; then
...
-    # Only install dirmngr if it's available in the cache
-    # it may not be available on Ubuntu <= 14.04 but it's not required there
-    cache_output=$(apt-cache search dirmngr)
-    if [ ! -z "$cache_output" ]; then
-      $sudo_cmd apt-get install -y dirmngr
+    apt_trusted_d_keyring="/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg"
+    apt_usr_share_keyring="/usr/share/keyrings/datadog-archive-keyring.gpg"
+
+    printf "\033[34m\n* Installing apt-transport-https, curl and gnupg\n\033[0m\n"
+    $sudo_cmd apt-get update || printf "\033[31m'apt-get update' failed, the script will not install the latest version of apt-tra    nsport-https.\033[0m\n"
```
Comment (KSerrania) [inline test/kitchen/site-cookbooks/dd-agent-install-script/files/dogstatsd_install_script.sh:246]:
```suggestion
    $sudo_cmd apt-get update || printf "\033[31m'apt-get update' failed, the script will not install the latest version of apt-transport-https.\033[0m\n"
```

**Review by KSerrania — APPROVED** [review]

**Comment by albertvaka** [comment]
Thanks for all those kitchen fixes! 😃 About this one: most of the logic in this dogstatsd install script is a copy-and-past of the linux install script. Could we maybe use the regular Linux install script passing `DD_AGENT_FLAVOR=datadog-dogstatsd` instead of maintaining this separate script? Currently that fails because it can't start the Agent service, but maybe it's easier to fix that than to maintain this script?

**Comment by KSerrania** [comment]
> Could we maybe use the regular install script passing DD_AGENT_FLAVOR=datadog-dogstatsd instead of maintaining this separate script? 

I looked into that yesterday, and yes that's doable. That will require a few more changes in the install script: the service name change you mentioned, the configuration directory & file used (https://github.com/DataDog/datadog-agent/blob/00aafcde7f12fd4e6f396f427ccbe962298b77c2/cmd/agent/install_script.sh#L15-L16), and maybe some other things I missed.

**Comment by bkabrda** [comment]
Yeah, it certainly is doable. I'll open a card for it :)

**Pull request merged**